### PR TITLE
Add option to use netcdf quantization

### LINKED ISSUE #1575: using HDF5 1.14.0 and Netcdf 4.9.2 in UFS to improve parallel netcdf compression feature
## Description
HDF5 version1.12.2 and above (latest release 1.14.0) have updates/bug fixes to improve parallel netcdf compression.  UFS  weather model will be updated to resolve some issues seens in the past.

## Solution
1) Need to install HDF5 1.14.0 and netcdf 4.9.2 on all the UFS WM supported platforms
2) update ufs UFS WM to use these new versions.

## Related to
library installation issues:

### LINKED ISSUE #1615: Testing the new netcdf 4.9.2 compression features in ufs-weather-model
## Description
New compression features are added in netcdf 4.9.0. UFS-weather-model will be updated to utilize the new features. 

## Solution
1) The new version of netcdf will be installed in the ufs weather model supported platforms 
2) The new compression feature with zstandard lib will be tested and used, replacing the current customized data quantization lossy compression algorithm used in FV3ATM IO.

### LINKED ISSUE #1929: transition to the build-in quantization capability in netcdf-c and clean up hard-coded quantization in fv3
## Description
This work will follow issue #1615 and #1928. We will use the build-in quantization capability in netcdf-c 4.9.2 and clean up quantization code in fv3.

One improvement from the build-in quantization in netcdf-c is that the bit grooming algorithm in the netcdf-c library will handle missing values. This helps resolve the issue we've seen in regional output domain with missing values when applying lossy compression. This helps us to resolve issue #1868.

### PR DESCRIPTION
**Pull request by DusanJovic-NOAA**
## PR Author Checklist:

- [x] I have linked PR's from all sub-components involved in section below. 
- [x] I am confirming reviews are completed in ALL sub-component PR's.
- [x] I have run the full RT suite on either Hera/Cheyenne AND have attached the log to this PR below this line:
  - LOG: 
[RegressionTests_hera.log](https://github.com/ufs-community/ufs-weather-model/files/13311988/RegressionTests_hera.log)

- [x] I have added the list of all failed regression tests to "Anticipated changes" section.
- [x] I have filled out all sections of the template.

## Description

This PR adds new configuration parameters which can be used to enable the netcdf quantization (lossy compression)
Two new (optional) model_configure parameters are: quantize_mode, quantize_nsd. The `quantize_mode` is a string which can have values 'quantize_bitgroom', 'quantize_granularbr' or 'quantize_bitround'., and `quantize_nsd` is number of significant digits. See documentation [here](https://docs.unidata.ucar.edu/netcdf-c/current/md__media_psf_Home_Desktop_netcdf_releases_v4_9_2_release_netcdf_c_docs_quantize.html).

### Associated UFSWM Issue to close

Closes #1929
Closes #1575
Closes #1615

### Subcomponent Pull Requests

NOAA-EMC/fv3atm/pull/718

### Subcomponents involved:
- [ ] AQM
- [ ] CDEPS
- [ ] CICE
- [ ] CMEPS
- [ ] CMakeModules
- [x] FV3
- [ ] GOCART
- [ ] HYCOM
- [ ] MOM6
- [ ] NOAHMP
- [ ] WW3
- [ ] stochastic_physics
- [ ] none

### Input data
- [x] No changes are expected to input data.
- [ ] Changes are expected to input data:
  - [ ] New input data.
  - [ ] Updated input data.

### Regression Tests:
- [ ] No changes are expected to any regression test.
- [x] Changes are expected to the following tests:

<details><summary>Tests effected by changes in this PR:</summary>

- control_wrtGauss_netcdf_parallel
- control_wrtGauss_netcdf_parallel_debug

</details>

### Libraries

- [x] Not Needed
- [ ] Needed
  - [ ] Create separate issue in [JCSDA/spack-stack](https://github.com/JCSDA/spack-stack) asking for update to library. Include library name, library version.
  - [ ] Add issue link from JCSDA/spack-stack following this item 

<details><summary>Code Managers Log</summary>

- [ ] This PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR.
- [ ] Move new/updated input data on RDHPCS Hera and propagate input data changes to all supported systems.
  - [ ] N/A

### Testing Log:
- RDHPCS
  - [ ] Hera
  - [ ] Orion
  - [ ] Hercules
  - [ ] Jet
  - [ ] Gaea
  - [ ] Cheyenne
- WCOSS2
  - [ ] Dogwood/Cactus
  - [ ] Acorn
- CI
  - [ ] Completed
- opnReqTest
  - [ ] N/A
  - [ ] Log attached to comment
</details>

### CODE REVIEW

**Comment by BrianCurtis-NOAA** [comment]
@DusanJovic-NOAA The RegressionTests_Hera.log you have shows comparison against your own REGRESSIONTESTS directory. Do you have one that shows the two tests failing against develop baselines?

**Comment by DusanJovic-NOAA** [comment]
[RegressionTests_hera.log](https://github.com/ufs-community/ufs-weather-model/files/13313932/RegressionTests_hera.log)

Hera log file against current baseline.

File: tests/default_vars.sh [hunk 1]
```diff
@@ -370,7 +370,7 @@ export OUTPUT_GRID="'cubed_sphere_grid'"
 export OUTPUT_FILE="'netcdf'"
 export ZSTANDARD_LEVEL=0
 export IDEFLATE=0
-export NBITS=0
+export QUANTIZE_NSD=0
```
Comment (junwang-noaa) [inline tests/default_vars.sh:375]: Since we will not use NBITS anymore, would you please create an issue on global-workflow to let them know this update?

Comment (DusanJovic-NOAA) [reply tests/default_vars.sh:375]: Opened NOAA-EMC/global-workflow#2053

**Comment by jkbk2004** [comment]
@DusanJovic-NOAA We can start working on this pr. If you are available, can you sync up branch?

**Comment by jkbk2004** [comment]
Regression tests run ok with this pr on RDHPCS sides. @BrianCurtis-NOAA @junwang-noaa FYI

**Comment by BrianCurtis-NOAA** [comment]
> Regression tests run ok with this pr on RDHPCS sides. @BrianCurtis-NOAA @junwang-noaa FYI

All set for WCOSS2 Systems

**Comment by zach1221** [comment]
The baselines create fine, but I continue to receive failures on Hercules when attempting to match against the atmf000.nc file, on Hercules.
image

**Comment by jkbk2004** [comment]
File are identical. It's nccmp hiccups. If I keep submitting those two failed cases, They run and compare ok. I think random system behavior.

**Comment by jkbk2004** [comment]
We will monitor the functionality of nccmp on hercules in coming PRs. BTW, we can move on to merge this pr.

**Comment by zach1221** [comment]
@DusanJovic-NOAA fv3atm PR #718 has been merged. Please revert changes to the git module url and update the fv3atm hash.
Hash: https://github.com/NOAA-EMC/fv3atm/commit/65c60173d420548380838ed93418c1c1b4c3c370

**Review by sadeghitabas — APPROVED** [review]

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Comment by DusanJovic-NOAA** [comment]
> The baselines create fine, but I continue to receive failures on Hercules when attempting to match against the atmf000.nc file, on Hercules. image

Can you please copy/paste text next time instead of screenshot image.

**Pull request merged**